### PR TITLE
KOGITO-9749: add addons deps to workflows to interact with external d…

### PR DESCRIPTION
**Pull request by nmirasch**
…ata index

jira: https://issues.redhat.com/browse/KOGITO-9749

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

File: serverlessworkflow/modules/ROOT/pages/data-index/data-index-service.adoc [hunk 1]
```diff
@@ -35,6 +35,34 @@ There are several ways to deploy the {data_index_ref} service. But there are som
...
+  <artifactId>kogito-addons-quarkus-events-process</artifactId> <1>
+</dependency>
+<dependency>
+  <groupId>org.kie.kogito</groupId>
+  <artifactId>kogito-addons-quarkus-process-management</artifactId> <2>
+</dependency>
+<dependency>
+  <groupId>org.kie.kogito</groupId>
+  <artifactId>kogito-addons-quarkus-source-files</artifactId> <3>
+</dependency>
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/data-index/data-index-service.adoc:57]:
```suggestion

```

We can remove this, as it's needed only for the management console. Sorry, I know that I'm adding for my JIRA, but it's unrelated.

File: serverlessworkflow/modules/ROOT/pages/data-index/data-index-service.adoc [hunk 2]
```diff
@@ -35,6 +35,34 @@ There are several ways to deploy the {data_index_ref} service. But there are som
...
+----
+
+<1> Allows the workflow to send the events that can be consumed by the {data_index_ref} service.
+<2> Creates the endpoints that allow management operations on {workflow_instances}.
+<3> Exposes the workflow definition sources.
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/data-index/data-index-service.adoc:62]:
```suggestion

```

**Review by ricardozanini — APPROVED** [review]

File: serverlessworkflow/modules/ROOT/pages/data-index/data-index-service.adoc [hunk 3]
```diff
@@ -35,6 +35,29 @@ There are several ways to deploy the {data_index_ref} service. But there are som
 . Provide the database connection properties, to allow data index store the indexed data. {data_index_ref} service does not initialize its database schema automatically. To initialize the database schema, you need to enable Flyway migration by setting QUARKUS_FLYWAY_MIGRATE_AT_START=true.
 . Define the `KOGITO_DATA_INDEX_QUARKUS_PROFILE` to set the way that the events will be connected (by default: `kafka-event-support`).
 
+[NOTE]
+====
+It is needed to prepare the {product_name} workflow to support the full communication with an external {data_index_ref} service.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/data-index/data-index-service.adoc:40]:
```suggestion
You must prepare the {product_name} workflow to support the full communication with an external {data_index_ref} service.
```

**Review by kaldesai — APPROVED** [review]
@nmirasch Just a minor suggestion. Otherwise, your content looks good to me. Thank you!

**Review by ricardozanini — APPROVED** [review]

**Review by radtriste — APPROVED** [review]

**Pull request merged**